torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2360,28 +2360,25 @@ Tensor slice_backward_wrapper(
 // https://j-towns.github.io/papers/svd-derivative.pdf
 //
 // This makes no assumption on the signs of sigma.
-Tensor svd_backward(const std::vector<torch::autograd::Variable> &grads, const Tensor& self,
-          bool some, bool compute_uv, const Tensor& raw_u, const Tensor& sigma, const Tensor& raw_v) {
-  TORCH_CHECK(compute_uv,
-           "svd_backward: Setting compute_uv to false in torch.svd doesn't compute singular matrices, ",
-           "and hence we cannot compute backward. Please use torch.svd(compute_uv=True)");
-
-  auto m = self.size(-2);
-  auto n = self.size(-1);
+Tensor svd_backward(const std::vector<torch::autograd::Variable> &grads,
+          bool full_matrices, const Tensor& U, const Tensor& sigma, const Tensor& Vh) {
+  auto m = U.size(-2);
+  auto n = Vh.size(-1);
   auto k = sigma.size(-1);
   auto gsigma = grads[1];
 
-  auto u = raw_u;
-  auto v = raw_v;
+  auto u = U;
+  auto v = Vh.mH();
   auto gu = grads[0];
-  auto gv = grads[2];
+  // grads[2] has the gradient wrt Vh
+  auto gv = grads[2].defined() ? grads[2].mH() : grads[2];
 
-  if (!some) {
+  if (full_matrices) {
     // We ignore the free subspace here because possible base vectors cancel
     // each other, e.g., both -v and +v are valid base for a dimension.
```

Comment:
"possible base vectors cancel each other" - I am not sure I understand this statement.

Reply:
This is there from before and I don't particularly like the writing either. The true statement should be:
>svd(A, full_matrices=True)  is not differentiable, but we avoid generating a gradient for the extra column vectors in it to push this problem under the rug.

Note how the tests do not test the value of the gradients on these tensors in the case `full_matrices=True`.

Reply:
Well, yes, because this is what we are doing. The cancelling effect does not sound right to me.

Reply:
You'll be happy to know that this comment (and the `full_matrices` argument to this function) is well gone in the next PR of the stack. This way people cannot get it wrong when using `svd_backward`, as they need to use already the thin matrices.